test/util.inc

```diff
@@ -17,23 +17,41 @@
 #
 #   $1 the directory containing the CMake simulation code
 function run_cmake_simulation() {
+  if [ -n "$ZSH_VERSION" ]; then
+    emulate -LR bash
+  fi
   log=$(mktemp)
-  trap "rm -rf \"${log}\"" RETURN
+  echo "created logfile ${log}"
+
+  if [ -n "$ZSH_VERSION" ]; then
+    TRAPEXIT() {
+      rm -rf "${log}"
+      sleep 1
+    }
+  else
+    trap "rm -rf \"${log}\"" RETURN
+  fi
 
   pushd "$1"
   biodynamo run | tee "${log}"
   SIM_RET_VAL=$?
   popd
 
   if [ "${SIM_RET_VAL}" != "0" ]; then
-    returm $SIM_RET_VAL
+    return $SIM_RET_VAL
   fi
 
   expected="Simulation completed successfully!"
   grep --quiet "${expected}" "${log}"
   return $?
 }
 
-BDM_PROJECT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+if [ -n "$BASH_VERSION" ]; then
```

Comment:
Can't we use $0 for both shell types?

Reply:
``$0`` unfortunately won't work if the [script is sourced](https://stackoverflow.com/a/35006505/13380539)